addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/SubResourceIntegrityAttributeScanRule.java

```diff
@@ -99,23 +111,64 @@ public static boolean contains(String tag) {
     @Override
     public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
 
+        SiteMap tree = Model.getSingleton().getSession().getSiteTree();
         List<Element> sourceElements = source.getAllElements();
         sourceElements.stream()
                 .filter(element -> SupportedElements.contains(element.getName()))
                 .filter(unsafeSubResource(msg.getRequestHeader().getHostName()))
                 .forEach(
-                        element ->
-                                newAlert()
-                                        .setRisk(Alert.RISK_MEDIUM)
-                                        .setConfidence(Alert.CONFIDENCE_HIGH)
-                                        .setDescription(getString("desc"))
-                                        .setSolution(getString("soln"))
-                                        .setReference(getString("refs"))
-                                        .setEvidence(element.toString())
-                                        .setCweId(345) // CWE-345: Insufficient Verification of Data
-                                        // Authenticity
-                                        .setWascId(15) // Application Misconfiguration
-                                        .raise());
+                        element -> {
+                            newAlert()
+                                    .setRisk(Alert.RISK_MEDIUM)
+                                    .setConfidence(Alert.CONFIDENCE_HIGH)
+                                    .setDescription(getString("desc"))
+                                    .setSolution(getString("soln"))
+                                    .setReference(getString("refs"))
+                                    .setEvidence(element.toString())
+                                    .setCweId(345) // CWE-345: Insufficient Verification of Data
+                                    // Authenticity
+                                    .setWascId(15) // Application Misconfiguration
+                                    .setOtherInfo(
+                                            Constant.messages.getString(
+                                                    MESSAGE_PREFIX + "otherInfo",
+                                                    calculateIntegrityHash(msg, element, tree)))
+                                    .raise();
+                        });
+    }
+
+    private String calculateIntegrityHash(HttpMessage msg, Element element, SiteMap tree) {
+        String src = element.getAttributeValue("src");
+        String integrityHash;
```

Comment:
You could just declare it null (or even better empty) then remove the else case, and assignments in the catch clauses.

Reply:
Yep, I did it in quick run. So I'll make the changes accordingly.